### ISSUE TITLE
Audio Recording Issue in audio_waveforms v1.0.5(Android)

### ISSUE THREAD

**Issue by Nehal-1234** [SimformSolutionsPvtLtd/audio_waveforms#381]
**Describe the bug**
When recording audio using the audio_waveforms package (version ^1.0.5), the recorded voice is not captured properly on Android devices. The recording includes unintended background noise or other external sounds that should not be part of the recording. This issue is not present on iOS devices, where the recording works as expected.

**To Reproduce**

Go to the part of the app where audio recording is implemented.

Tap on the 'Record' button to start recording audio.

Speak into the device's microphone.

Stop the recording and play it back.

Notice that the recorded audio contains additional background sounds or unintended noise on Android devices.

**Expected Behavior**
The recorded audio should only capture the user's voice without any additional noise or interference. This behavior is already achieved on iOS devices but not on Android.

**Smartphone (please complete the following information):**

Device: Samsung Galaxy S23, Vivov2025, OPPO F15

OS: [Android 11 to 14]

Version: [ audio_waveforms v1.0.5]

**Screenshots**
see the difference between [Android](https://drive.google.com/file/d/1cDM1qeaPRAUsCfZCm1-ITq-xYIQAmXKb/view) and [iOS](https://drive.google.com/file/d/1EYLqvp6g_Ty3AdoRotrHc9e7-By_idz5/view) :

**Additional Context**
This issue is specific to Android devices and does not affect iOS devices.

**Comment by jay-simformsolutions** [comment]
Hi @Nehal-1234, can you please increase the bitrate and check that the noise is still in the recording sound?

**Comment by Nehal-1234** [comment]
Yes, @jay-simformsolutions, I increased the bitrate, but the noise issue persists. I also tested it with version 1.1.0, but there was no improvement. The app is live, and we are receiving complaints from Android users. Can you please help resolve this issue?

**Comment by jay-simformsolutions** [comment]
@Nehal-1234, can you please provide us the value of bit rate you have set?

**Comment by Nehal-1234** [comment]
@jay-simformsolutions ,here is the bitRate = 256000 and also I tried with 512000

**Comment by jay-simformsolutions** [comment]
Hi @Nehal-1234, We currently record audio using MediaRecorder. However, to add noise suppression functionality, we need to change the class from MediaRecorder to AudioRecord, which will require more effort. 

But we will take your issue as an enhancement and add a noise suppression feature as well. 

Thanks for filling this issue.

**Comment by lavigarg-simform** [comment]
Hi @Nehal-1234! Thanks for your patience. Since v2.0.0 the Android recorder uses `AudioRecord` instead of `MediaRecorder` (latest release is 2.0.2). On top of that, I've linked PR #499, which adds options for noise suppression, echo cancellation, and automatic gain control.

These effect options aren't in a release yet — could you test from the linked PR with the settings below and let me know if it resolves the noise?

```dart
RecorderSettings(
  androidEncoderSettings: AndroidEncoderSettings(
    audioSource: AndroidAudioSource.voiceCommunication,
    useNoiseSuppressor: true,
    useEchoCanceler: true,
  ),
)
```

Note: these effects are attached only when the device supports them, so results can vary by hardware, worth testing on the affected Samsung/Vivo/OPPO devices. Thanks again for reporting and for your continued support of the package!
